app/models/user.rb

```diff
@@ -72,4 +78,14 @@ def new_token
   def downcase_email
     email.downcase!
   end
+
+  def password_required?
+    password_digest.blank? || !password.nil?
+  end
+
+  def password_presence_if_confirmation_provided
+    return unless password.blank? && password_confirmation.present?
```

Comment:
```suggestion
    return unless password.blank? && password_confirmation.present?
```
Mệnh đề 2 đk trở lên thì e dùng if cho tường mình nhé